### ISSUE TITLE
Rewrite logic for consistent checking of required and optional BibTex fields

### ISSUE THREAD

**Issue by stefan-kolb** [JabRef/jabref#316]
Minimal example:
```
@Article{bad2012cloud,
  Title                    = {{Cloud Computing Synopsis and Recommendations}},
  Author                   = {Badger, Lee and others},
  Journal                  = {NIST Special Publication 800-146},
  Year                     = {2012}
}
```

**Comment by stefan-kolb** [comment]
Required: IEEE: http://ctan.mirrors.hoobly.com/macros/latex/contrib/IEEEtran/bibtex/IEEEtran_bst_HOWTO.pdf

**Comment by stefan-kolb** [comment]
article An article from a journal or magazine. Required fields: author, title,
journal, year. Optional fields: volume, number, pages, month, note.
book A book with an explicit publisher. Required fields: author or editor,
title, publisher, year. Optional fields: volume or number, series,
address, edition, month, note.
booklet A work that is printed and bound, but without a named publisher or
sponsoring institution. Required field: title. Optional fields: author,
howpublished, address, month, year, note.
conference The same as INPROCEEDINGS, included for Scribe compatibility.
inbook A part of a book, which may be a chapter (or section or whatever)
and/or a range of pages. Required fields: author or editor, title,
chapter and/or pages, publisher, year. Optional fields: volume or
number, series, type, address, edition, month, note.
incollection A part of a book having its own title. Required fields: author,
title, booktitle, publisher, year. Optional fields: editor, volume or
number, series, type, chapter, pages, address, edition, month, note.
8
inproceedings An article in a conference proceedings. Required fields:
author, title, booktitle, year. Optional fields: editor, volume or
number, series, pages, address, month, organization, publisher,
note.
manual Technical documentation. Required field: title. Optional fields:
author, organization, address, edition, month, year, note.
mastersthesis A Master’s thesis. Required fields: author, title, school,
year. Optional fields: type, address, month, note.
misc Use this type when nothing else fits. Required fields: none. Optional
fields: author, title, howpublished, month, year, note.
phdthesis A PhD thesis. Required fields: author, title, school, year. Optional
fields: type, address, month, note.
proceedings The proceedings of a conference. Required fields: title, year.
Optional fields: editor, volume or number, series, address, month,
organization, publisher, note.
techreport A report published by a school or other institution, usually numbered
within a series. Required fields: author, title, institution, year.
Optional fields: type, number, address, month, note.
unpublished A document having an author and title, but not formally published.
Required fields: author, title, note. Optional fields: month,
year.

**Comment by stefan-kolb** [comment]
- [ ] our Bibtex types are not complete
- [ ] We have multiple types that are not known to BibTex

**Comment by oscargus** [comment]
Actually, IEEEtran only require those four fields as well (as stated in the PDF). 

The four additional ones are (as you probably noted) from IEEEtran. I'd say that at least patent and standard are really useful. ieeetranbstctl is for controlling some things when using IEEEtran and without it, the entry will be deleted everytime the .bib-file is saved rendering JabRef rather useless when publishing in IEEE...

**Comment by koppor** [comment]
Regarding the bibtex types: They are defined by the `bst` files. I think the team around [biblatex](https://www.ctan.org/pkg/biblatex) thought intensively about "good" types. Maybe, we should align with biblatex? If we want to support bsts by the publishers, we should go through the [LNCS style](https://github.com/latextemplates/LNCS), the IEEE style etc. and collect the entry types available there. Possibly collecting them in a Wiki page.

**Comment by stefan-kolb** [comment]
Imho, we should separate types from custom styles from the default Bibtex entry types. 
At least for the sake of good separation. Anyhow, how will entry types behave if used with styles that don't support these?

**Comment by stefan-kolb** [comment]
Implemented in #352.
